### PR TITLE
Add some automation tools

### PR DESCRIPTION
**Pull request by rmohr**
## release-querier

`release-querier` is a tool to find latest releases on github projects which use semver.

`release-querier` does the following:

1. Finding the latest release of a github project
2. Finding the latest patch release of a minor release
3. Finding the last three minor releases of a major release (what k8s supports)
4. Format the output via a configurable template for better script integration

## Example

```
$ # Latest k8s release
$ release-querier --github-token-path="" -org=kubernetes -repo=kubernetes -latest
v1.17.3
$
$ # Current latest supported k8s versions
$ release-querier --github-token-path="" -org=kubernetes -repo=kubernetes -last-three-minor-of v1
v1.17.3
v1.16.7
v1.15.9
$
$ # Latest 1.17 patch release
$ release-querier --github-token-path="" -org=kubernetes -repo=kubernetes -last-patch-of v1.17
v1.17.3
$ # Change the output format
$ release-querier --github-token-path="" -org=kubernetes -repo=kubernetes -last-three-minor-of v1 --template="{{.Major}}.{{.Minor}}"
1.17
1.16
1.15
```

## kubevirtci-bumper

`kubevirtci-bumper` is a tool to automatically bump kubevirtci providers.

`kubevirtci-bumper` does the following:

1. Ensuring that we have a provider for the latest minor release by copying a predecessor provider
2. Ensuring that the last three minor releases are up-to-date
3. Not yet implemented: Dropping unsuppored k8s providers automatically
4. Not yet implemented: Cloning and Dropping correspnding prow jobs
5. Not yet implemented: Create PRs agains kubevirtci and project-infra

## Example

```
$ # Ensure that we have the latest provider in kubevirtci
$ kubevirtci-bumper --github-token-path="" --ensure-latest --k8s-provider-dir /go/src/kubevirt.io/kubevirtci/cluster-provision/k8s
  {"level":"info","msg":"Added provider 1.17 with version 1.17.3","time":"2020-02-19T11:46:15+01:00"}
  {"level":"info","msg":"Minor version 1.17 updated to 1.17.3","time":"2020-02-19T11:46:15+01:00"}
$ release-querier --github-token-path="" -org=kubernetes -repo=kubernetes -latest
v1.17.3
$
$ # Ensure that we point to the latest patch release of the last three supported minor k8s releases
$ kubevirtci-bumper --github-token-path="" --ensure-last-three-minor-of v1 --k8s-provider-dir /go/src/kubevirt.io/kubevirtci/cluster-provision/k8s
  {"level":"info","msg":"Minor version 1.17 updated to 1.17.3","time":"2020-02-19T11:47:17+01:00"}
  {"level":"info","msg":"Minor version 1.16 updated to 1.16.7","time":"2020-02-19T11:47:17+01:00"}
  {"level":"info","msg":"Minor version 1.15 updated to 1.15.10","time":"2020-02-19T11:47:17+01:00"}
```

### CODE REVIEW

File: robots/release-querier/main.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.org, "org", "kubevirt", "Organization")
+	fs.StringVar(&o.repo, "repo", "kubevirt", "Organization")
+	fs.StringVar(&o.TokenPath, "github-token-path", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	fs.StringVar(&o.endpoint, "github-endpoint", "https://api.github.com/", "GitHub's API endpoint (may differ for enterprise).")
+	fs.BoolVar(&o.latest, "latest", false, "Query for the latest release")
+	fs.StringVar(&o.latestThreeMinor, "last-three-minor-of", "", "Query for the last three minor releases of a given release (e.g. v1 or 2)")
```
Comment (qinqon) [inline robots/release-querier/main.go]: Three would be enough right ?

Comment (rmohr) [reply robots/release-querier/main.go]: could you rephrase what you mean?
